### ISSUE TITLE
Deceptive Site Ahead

### ISSUE THREAD

**Issue by GodBleak** [jellyfin/jellyfin-web#4076]
**Describe The Bug**
A domain hosting Jellyfin is flagged by Google as a "Deceptive Site".

**Steps To Reproduce**
Unknown

**System (please complete the following information):**
 - Browser: Firefox, Chrome
 - Jellyfin Version: 10.8.5 (linuxserver/jellyfin:10.8.5-1-ls180)

**Additional Context**
Google claims that https://example.tld/web/index.html
> attempts to trick users into doing something dangerous, such as installing unwanted software or revealing personal information.

I've appealed to Google twice now, but the domain continues to be flagged.
This issue has been further documented on a few reddit posts:
- [Deceptive Site Ahead](https://www.reddit.com/r/jellyfin/comments/xqk01i/deceptive_site_ahead)
- ["Deceptive site ahead" warning on my self hosted jellyfin web app](https://www.reddit.com/r/selfhosted/comments/xq162z/deceptive_site_ahead_warning_on_my_self_hosted/)

**Comment by thornbill** [comment]
Does entering your url here provide any information about what they believe is an issue? https://transparencyreport.google.com/safe-browsing/search

We really have nothing to go off of for this currently.

**Comment by GodBleak** [comment]
Unfortunately, this is all it says

> Current status warning
> This site is unsafe
> 
> The site https://example.tld/web/index.html contains harmful content, including pages that:
> 
> - Try to trick visitors into sharing personal info or downloading software

**Comment by thornbill** [comment]
Are you using any third party css?

**Comment by lednerg** [comment]
The same thing just happened to me tonight. My server's been using the same IP (from Comcast) for at least a couple years now. I'm currently on version 10.8.1 and am not using any third party CSS. I have the following plugins installed: [(PNG of plugins page)](https://i.imgur.com/BXCiVPd.png).

**Comment by GodBleak** [comment]
Sorry @thornbill, was only just notified  of updates on the thread, no I'm not using any third-party CSS

**Comment by mcshaman** [comment]
Same issue here!

**Comment by viletuna** [comment]
I'm also having the same issue. Twice now with two different servers. Both were using duck DNS and caddyv2. Requesting Google to remove the flag worked temporarily before being flagged again

**Comment by GodBleak** [comment]
I've done a bit of digging. It seems the YunoHost community is also [experiencing this](https://forum.yunohost.org/t/google-flags-my-sites-as-dangerous-deceptive-site-ahead/20361/94). With further digging I found a few things that leads me to suspect our domains are being flagged for "Insufficiently labeled third-party services". 

1. While not directly related to Safe Browsing (and thus this error), I found this [notice](https://www.reddit.com/r/Bitwarden/comments/t4n84j/comment/hz5udac/?context=3) from NameCheap: 

> >  Please be informed that the xxxxxx domain name was reported as involved in abusive activity by a trusted organization. During the investigation, it was noticed that your website content is a copy of the Bitwarden official website. On that ground, we were forced to suspend the domain name due to phishing activities, which include unauthorized use of the legitimate organization denomination and attempts to acquire sensitive information such as usernames, passwords, etc

And they follow that up with:
> > you will need to provide us with paperwork proving your cooperation with the Bitwarden website and their consent to use their official denomination in your domain name.

Indicating that the action they're taking is to recognize IP(intellectual property) violations used to phish. Furthermore, it implies the action is automatic and not at the request of the maintainer (Bitwarden)

2. [This comment](https://www.reddit.com/r/linux/comments/8x75f0/comment/e21lr3o?context=3) regarding the Deceptive Site warning also seems to indicate that this is more of a branding/IP problem, rather than just an issue with the source code. 

3. And [this comment](https://stackoverflow.com/a/68468475) on StackOverflow where someone supposedly received the notice on a site imitating Netflix, also believes that the issue is a result of the imitation.

4. Eventually, I found this article by Google on [social engineering](https://developers.google.com/search/docs/monitor-debug/security/social-engineering) where they show [deceptive content examples](https://developers.google.com/search/docs/monitor-debug/security/social-engineering#deceptive-content-examples) 

This one caught my eye.
![](https://lh3.googleusercontent.com/uTWvxf7NR7alzT_VEF1wD31v3l6CeKG7M7nN4wpa-Z_nSGb5xsMUi19RTUTKQimqbu0=w577)

Its layout is similar to the JellyFin login page, right? A page at the root path of a domain (true for both the OP of the YunoHost thread and myself) using a trusted third-party's logo in an authoritative position, with the page's sole purpose clearly being to collect credentials. YunoHost shares this [layout as well](http://thefeaturecreep.com/). Additionally, both apps use the product name in the page title, along with the product's logo as the favicon.

So I think the combination of

- the page title being "JellyFin"
- the page favicon using the JellyFin logo.
- the authoritative location of the JellyFin logo
- the page's sole purpose being to collect credentials, and
- the service being hosted at the root path of the FQDN

leads to Google thinking we're trying to impersonate JellyFin.

**Comment by mcshaman** [comment]
Interesting hypothesis @GodBleak. Do you know if it is possible to override all these on the landing page?

**Comment by thornbill** [comment]
I suspect the meta tags here may be to blame, but someone would have to test that to verify since Google is providing no usable information.

https://github.com/jellyfin/jellyfin-web/blob/master/src/index.html#L15-L19

**Comment by NeonWizard** [comment]
I disputed the "deceptive site warning" through the Google search console about a week ago, and the error has yet to come back.

**Comment by VTStation** [comment]
> I disputed the "deceptive site warning" through the Google search console about a week ago, and the error has yet to come back.

I'v had this issue since mid of sept , lodge a review to google via search console ,they would lift the block and then aweek later it will be blocked again.  I'v been blocked 4 times , rebuilt the server the first time after finding no issues , they still blocked it and i have continued to send them the same review response " please stop blocking this private site "  they have lifted the block every time.. Iv stopped sending reviews to google 'i gave up ..  using jellyfin in kodi app is my work around .

**Comment by lednerg** [comment]
> I suspect the meta tags here may be to blame, but someone would have to test that to verify since Google is providing no usable information.
> 
> https://github.com/jellyfin/jellyfin-web/blob/master/src/index.html#L15-L19

Twelve days ago I changed all five of those meta tags in my jellyfin-web\index.html file so that they're all unique to my server and I have yet to be blocked by Google again. I've logged in and out remotely several times since then using Google devices/programs. I'm not saying I'm sure this is definitely a fix, I'm just sharing my experience. BTW, editing that file was a pain since it's all on one line.

**Comment by VTStation** [comment]
> > I suspect the meta tags here may be to blame, but someone would have to test that to verify since Google is providing no usable information.
> > https://github.com/jellyfin/jellyfin-web/blob/master/src/index.html#L15-L19
> 
> Twelve days ago I changed all five of those meta tags in my jellyfin-web\index.html file so that they're all unique to my server and I have yet to be blocked by Google again. I've logged in and out remotely several times since then using Google devices/programs. I'm not saying I'm sure this is definitely a fix, I'm just sharing my experience. BTW, editing that file was a pain since it's all on line.


I also have qbittorrent web server running and that is blocked by google its not limited to jellyfin, Alot of people are running 

> > I suspect the meta tags here may be to blame, but someone would have to test that to verify since Google is providing no usable information.
> > https://github.com/jellyfin/jellyfin-web/blob/master/src/index.html#L15-L19
> 
> Twelve days ago I changed all five of those meta tags in my jellyfin-web\index.html file so that they're all unique to my server and I have yet to be blocked by Google again. I've logged in and out remotely several times since then using Google devices/programs. I'm not saying I'm sure this is definitely a fix, I'm just sharing my experience. BTW, editing that file was a pain since it's all on one line.

Are you able to compare from previous versions of the jellyfin server ? if this tag had changed after the update causing google block? as iv been running jellyfin for a few years with no issues up until now
.

**Comment by lednerg** [comment]
> Are you able to compare from previous versions of the jellyfin server ? if this tag had changed after the update causing google block? as iv been running jellyfin for a few years with no issues up until now

I found some older versions of the index.html file going back to last November and those meta tags haven't changed. This is something new that Google has started doing all of the sudden. I'm just a layman but I looked into what those "og" (Open Graph) tags are about and it appears that people have done phishing scams using false og tags as a way to trick people into thinking they're logging into their bank or whatever.

If this is actually what the problem is - and we don't know yet - then that would mean Google sees that your Jellyfin server has an "og:url" tag pointing to "https://jellyfin.org" - but that isn't your server's URL, so Google may be assuming you're trying to spoof people. Again, we don't know if that's what's going on. FWIW, I changed my "og:url" to my server's IP address, and changed "og:title", "og:name", and "og:description" to "lednerg's Jellyfin Server".

**Comment by VTStation** [comment]
> > Are you able to compare from previous versions of the jellyfin server ? if this tag had changed after the update causing google block? as iv been running jellyfin for a few years with no issues up until now
> 
> I found some older versions of the index.html file going back to last November and those meta tags haven't changed. If the tags are what the issue is, then this is something new that Google has started doing all of the sudden. I'm just a layman but I looked into what those "og" (Open Graph) tags are about and it appears that people have done phishing scams using false og tags as a way to trick people into thinking they're logging into their bank or whatever.
> 
> If this is actually what the problem is - and we don't know yet - then that would mean Google sees that your Jellyfin server has an "og:url" tag pointing to "https://jellyfin.org" - but that isn't your server's URL, so Google may be assuming you're trying to spoof people. Again, we don't know if that's what's going on. FWIW, I changed my "og:url" to my server's IP address, and changed "og:title", "og:name", and "og:description" to "lednerg's Jellyfin Server".

I changed those og tags and guess what ", google blocked the site :/ . Is it because google detected change ?.. I'll send a review to get it unblocked with this current change , see how long it remains unblocked .

**Comment by lednerg** [comment]
It could be that you didn't change them soon enough, but like I said, we don't actually know what the problem is.

After my server was blocked by Google, I turned it off, got my IP unblocked, and temporarily switched to using an Apache server. I only turned Jellyfin back on after changing those meta tags. That was 16 days ago and my server hasn't been blocked since. I've been accessing the server from outside of my local network practically every day, in ways which would be going through Google Security, such as through Chrome browsers and Android devices. Unfortunately, I can't revert the tags back just to test if it'll block me again because I'm using this IP (and Jellyfin) for work; I use it to serve videos I make for my clients.

**Comment by optroodt** [comment]
I'm facing the same issue, requested to be reviewed once after which the warnings disappeared, only to return a few days later.
I went through the verification process on https://search.google.com/search-console, and then this caught my eye:
![image](https://user-images.githubusercontent.com/1515592/202033231-17783799-2108-440f-a0fe-fab06bcf16e6.png)

Could it be because of the service workers that Jellyfin uses? Maybe in combination with the `og:url` tag and asking for login details?

**Comment by VTStation** [comment]
this is what is on mine.

![image](https://user-images.githubusercontent.com/117891637/202176535-027eb8ae-1d58-45ab-b7e4-c8d87f955a1a.png)

**Comment by Hukuma1** [comment]
Battled with this earlier. Took down my whole domain. Luckily disputing it seemed to have corrected it. Not happy to read it can still happen after, and multiple times no less...

**Comment by lednerg** [comment]
After three weeks or so with it being fine, Google has flagged my server again. I have no idea what to do, but I obviously can't use Jellyfin anymore. Just wrote a detailed saga to Google about it, but who knows if that'll even reach a conscious human.

**Comment by VTStation** [comment]
> After three weeks or so with it being fine, Google has flagged my server again. I have no idea what to do, but I obviously can't use Jellyfin anymore. Just wrote a detailed saga to Google about it, but who knows if that'll even reach a conscious human.

Same here just got blocked that didn't last long, so the tag mod did not do anything :/

**Comment by VTStation** [comment]
Google still flagging site, but strangely Android apps are working ... Anyone else experiencing this to ? .. maybe google has made an exception ?

**Comment by optroodt** [comment]
FYI Without doing anything to Jellyfin, Safari no longer displays the warning for my domain, but Chrome still does.

**Comment by mike948** [comment]
I got the same warning a month ago. Afterwards I added the domain to Google Search Console and filed a review. Within a couple days they removed the warning. I just got a new email from Google Search Console saying "Social engineering content detected on `<mydomain.tld>`" and the warning is back. It says the deceptive page is `https://mydomain.tld/web/index.html`

Details about my setup: Running Jellyfin in Docker with Nginx Proxy Manager and cloudflare-ddns. Additionally have the Cloudflare DNS proxy status enabled and Cloudflare's Web Application Firewall setup to block all access outside the USA.

**Comment by VTStation** [comment]
> I got the same warning a month ago. Afterwards I added the domain to Google Search Console and filed a review. Within a couple days they removed the warning. I just got a new email from Google Search Console saying "Social engineering content detected on `<mydomain.tld>`" and the warning is back. It says the deceptive page is `https://mydomain.tld/web/index.html`
> 
> Details about my setup: Running Jellyfin in Docker with Nginx Proxy Manager and cloudflare-ddns. Additionally have the Cloudflare DNS proxy status enabled and Cloudflare's Web Application Firewall setup to block all access outside the USA.

There is no fix ,. Google never ending flag , lately apps on tv and app still works which is all that matters..

**Comment by devopstagon** [comment]
I suspect it could be reverse proxy settings. This block seems to cause issues:

```
    # location block for /web - This is purely for aesthetics so /web/#!/ works instead of having to go to /web/index.html/#!/
    location = /web/ {
        # Proxy main Jellyfin traffic
        proxy_pass http://$jellyfin:8096/web/index.html;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Protocol $scheme;
        proxy_set_header X-Forwarded-Host $http_host;
    }
```

At least this is my initial suspicion.

**Comment by VTStation** [comment]
> 

Iv experience with and without reverse proxy .. google just flag regardless.

**Comment by candry7731** [comment]
this seems like it worked for me...... I have gone 2 weeks without being flagged again, after I changed the tags

**Comment by misterkiem** [comment]
I got flagged last Sunday (2022.12.25) and after reading about this issue I immediately submitted a review request on google search console. flag was removed on Wednesday (2022.12.28), and then just auto flagged again today (2022.12.31). I have just updated the og: meta tags as suggested here and will report results in a few weeks

**Comment by candry7731** [comment]
Yep, I changed the meta tags and got tagged again........

Not sure whatelse to do.

**Comment by GodBleak** [comment]
I've dug into the Safe Browsing project a bit and couldn't find any information on how projects such as jellyfin are supposed to deal with this, couldn't find any formal contact information for the project either, the only contact I could find is [@loicberton](https://github.com/loicbertron) -- a Engineering Manager for the project (whom tastefully has the notice as his [cover photo](https://infosec.exchange/@loic) on mastodon), I sent him an email a month or so back bringing this to his attention and got nothing but radio silence back.  

Considering there's no real course of action to actually stay off their list, at this point I'd call the Safe Browsing project a malicious and thinly veiled attempt by Google to control the wider internet under the guise of "protecting the public". So much for "don't be evil"

**Comment by nimids** [comment]
My Jellyfin server just got hit with this, it flagged all of my subdomains. 
Added the domain to Google Search Console and requested review. 
Really hoping it gets unflaged, as I use the domain for many other things. 
Then I'm going to enable basic HTTP auth on my reverse proxy in a hope it will prevent reflaging. 
As without a password they will only get 401 Unauthorized. 
Unfortunately, I don't think the non web clients support that.

**Comment by candry7731** [comment]
@nimids make sure you change the html meta tags. Then submit for review. 

I had to submit for review 1 more time after I changed the meta tags and have not been flagged again. I was flagged 4 times in total.

**Comment by misterkiem** [comment]
Just updating from [this comment](https://github.com/jellyfin/jellyfin-web/issues/4076#issuecomment-1368299377). After updating the meta tags I resubmitted for review and the flag was lifted within 2 hours. It has been 3 and a half weeks since then with no problems.

@nimids the process I specifically followed was this:

1. Edit index.html in the jellyfin web folder (path to this can be found in the admin dashboard at the bottom)
2. Search for "og:", it will highlight a bunch of meta properties.
3. Change all of them (og:title, og:site_name, og:url, og:description. og:type can be left on 'article')
4. Resubmit your site for review. In the comments for the review I specifically mentioned that I changed the og: meta properties to something unique from jellyfin.org

**Comment by seclusionenthusiast** [comment]
A good alternative to changing the meta tags may be to just link this thread when requesting a review. I pointed out this was a well-documented issue with Jellyfin and they lifted my flag within a couple of hours.

**Comment by OdinGitDat** [comment]
I really don't think it matters what you write when requesting a review. I left the textbook practically empty and the flag was lifted after 2 hours. 

I believe speculating here about what works is pointless without any proof or confirmation from someone at Google.

**Comment by christovic** [comment]
I didn't want to have to worry about changing the tags so I've done this, hope it helps someone.

I've created `Dockerfile` next to my `docker-compose.yaml` with the following content:

Feel free to change "My Jellyfin" and "https://jellyfin.example.org" to whatever you want.

```
FROM jellyfin/jellyfin:10.8.9

RUN sed -i -e 's|<meta property="og:title" content="Jellyfin">|<meta property="og:title" content="My Jellyfin">|' \
    -e 's|<meta property="og:site_name" content="Jellyfin">|<meta property="og:site_name" content="My Jellyfin">|' \
    -e 's|<meta property="og:url" content="http://jellyfin.org">|<meta property="og:url" content="https://jellyfin.example.org">|' \
    -e 's|<meta property="og:description" content="The Free Software Media System">|<meta property="og:description" content="Private Media Server">|' \
```

and my docker-compose.yaml from:
```
image: jellyfin/jellyfin:10.8.9
```
to:
```
build: .
```

**Comment by alanmccallum** [comment]
I was using jellyfin.example.tld

Prior to being flagged I was using Ngnix Proxy Manager with a simple Let's Encrypt cert for the subdomain.

Last week my Ngnix Proxy Manager install became corrupted, so I set up a new install. After the reinstall of NPM I used Let's Encrypt along with a DNS challenge to GoDaddy with a wildcard for the subdomains.

My entire domain was flagged as a deceptive site a day later. I disputed with google with no change.

I updated the tags in Jellyfin as previously documented, and used a fresh domain - video.otherexample.tld - using the same NPM and GoDaddy account. The new domain was flagged within a couple days.

This may be a coincidence, but the timing is odd.

**Comment by harrv** [comment]
> I've done a bit of digging. It seems the YunoHost community is also [experiencing this](https://forum.yunohost.org/t/google-flags-my-sites-as-dangerous-deceptive-site-ahead/20361/94). With further digging I found a few things that leads me to suspect our domains are being flagged for "Insufficiently labeled third-party services".
> 
>     1. While not directly related to Safe Browsing (and thus this error), I found this [notice](https://www.reddit.com/r/Bitwarden/comments/t4n84j/comment/hz5udac/?context=3) from NameCheap:
> 
> 
> > > Please be informed that the xxxxxx domain name was reported as involved in abusive activity by a trusted organization. During the investigation, it was noticed that your website content is a copy of the Bitwarden official website. On that ground, we were forced to suspend the domain name due to phishing activities, which include unauthorized use of the legitimate organization denomination and attempts to acquire sensitive information such as usernames, passwords, etc
> 
> And they follow that up with:
> 
> > > you will need to provide us with paperwork proving your cooperation with the Bitwarden website and their consent to use their official denomination in your domain name.
> 
> Indicating that the action they're taking is to recognize IP(intellectual property) violations used to phish. Furthermore, it implies the action is automatic and not at the request of the maintainer (Bitwarden)
> 
>     2. [This comment](https://www.reddit.com/r/linux/comments/8x75f0/comment/e21lr3o?context=3) regarding the Deceptive Site warning also seems to indicate that this is more of a branding/IP problem, rather than just an issue with the source code.
> 
>     3. And [this comment](https://stackoverflow.com/a/68468475) on StackOverflow where someone supposedly received the warning on a site imitating Netflix, also believes that the issue is a result of the imitation.
> 
>     4. Eventually, I found this article by Google on [social engineering](https://developers.google.com/search/docs/monitor-debug/security/social-engineering) where they show [deceptive content examples](https://developers.google.com/search/docs/monitor-debug/security/social-engineering#deceptive-content-examples)
> 
> 
> This one caught my eye. ![](https://camo.githubusercontent.com/ac67014f8e8d3554460b3d39421d44032be81d7c90bf09767a8589b0685269bf/68747470733a2f2f6c68332e676f6f676c6575736572636f6e74656e742e636f6d2f755457767866374e5237616c7a545f564546317744333176336c3643654b47374d376e4e347770612d5a5f6e5347623578734d55693139525455544b51696d716275303d77353737)
> 
> Its layout is similar to the JellyFin login page, right? A page at the root path of a domain (true for both the OP of the YunoHost thread and myself) using a trusted third-party's logo in an authoritative position, with the page's sole purpose clearly being to collect credentials. YunoHost shares this [layout as well](http://thefeaturecreep.com/). Additionally, both apps use the product name in the page title, along with the product's logo as the favicon.
> 
> So I think the combination of
> 
>     * the page title being "JellyFin"
> 
>     * the page favicon using the JellyFin logo
> 
>     * the authoritative location of the JellyFin logo
> 
>     * the page's sole purpose being to collect credentials, and
> 
>     * the service being hosted at the root path of the FQDN
> 
> 
> leads to Google thinking we're trying to impersonate JellyFin.

After reading this issue and all of the comments up to now, I'd say this comment by GodBleak appears to be the most likely cause of our personal servers getting flagged. Google bots assume that our Jellyfin instances are all trying to impersonate https://demo.jellyfin.org/ because the login pages are all nearly identical (including elements like favicon and the login page title, Jellyfin). 

Short of convincing Google that the users of our sites are not being tricked into giving up their login credentials for demo.jellyfin.org, the best way to avoid getting our servers flagged as deceptive is to change as many of the common elements as we can. Jellyfin devs could help with this by providing a robust customization feature that allows elements such as page titles and all branding images (including favicon) to be changed by the Jellyfin admin.

By the way, Emby users are also plagued by this lately (same issue there with Emby instances appearing to impersonate https://app.emby.media).

**Comment by FastThenLeft** [comment]
My site flagged as "Deceptive" yet again. After reading TONS and coming across the same article as @harrv above, I agree.

The site is here if you want to read it:
https://developers.google.com/search/docs/monitor-debug/security/social-engineering

**The site specifically has information for sites using a "third-party service"**. Maybe that would be helpful:
https://developers.google.com/search/docs/monitor-debug/security/social-engineering#third-party-guidelines

**Comment by FastThenLeft** [comment]
I decided to put the actual Google "third-party service guidelines" right here:

Third-party service guidelines
If you include a third-party service in your site, we recommend that you meet the following conditions in order to avoid being labeled as social engineering:

- On every page, the third-party site clearly includes the third-party brand in a way that ensures users understand who is operating the site. For example, by including the third-party brand at the top of the page.
- On every page that contains first-party branding, explicitly state the relationship between the first and third party, and provide a link for more information. For example, a statement like this:This service is hosted by Example.com on behalf of Example.charities.com. More information.

A good usability guideline is whether a user viewing the page in isolation understands which site they are on, and the relationship between the first and third party at all times.

**Comment by FastThenLeft** [comment]
I decided to put the Google "Third-party service guidelines" right here so you don't have to go anywhere to read them:

> **Third-party service guidelines**
> If you include a third-party service in your site, we recommend that you meet the following conditions in order to avoid being labeled as social engineering:
> 
> - On every page, the third-party site clearly includes the third-party brand in a way that ensures users understand who is operating the site. For example, by including the third-party brand at the top of the page.
> - On every page that contains first-party branding, explicitly state the relationship between the first and third party, and provide a link for more information. For example, a statement like this:
> This service is hosted by Example.com on behalf of Example.charities.com. More information.
> 
> A good usability guideline is whether a user viewing the page in isolation understands which site they are on, and the relationship between the first and third party at all times.

**Comment by mcshaman** [comment]
I shut down my jellyfin server and put in a request with Google (through search console) to unflag my domain about 4 days ago but it is still flagged. How long does the typically take? Does Google communicate the progress of the process?

**Comment by FastThenLeft** [comment]
Hours for me.

**Comment by nimids** [comment]
I removed basic HTTP auth on my reverse proxy to use a non-web client got near instantly got a message about Social engineering and a deceptive page listed on Google Search Console.

**Comment by morkyy** [comment]
I just started  getting the exact same issue last night. Pretty bummed that there is no fix yet as this has been ongoing for over 6 months.

**Comment by chenichadowitz** [comment]
I ran into this over the last few weeks and tried to push the support team on what exactly triggered the flag. I also pointed them to this issue. After some back and forth, this is the result I got from them:

>We received an update from our team. We cannot provide further details on how our system detect suspicious pages from your website. Please ensure your website follows the guidelines laid out on including a third-party service on your website to make it clear about the relationship between your site and the third-party. https://developers.google.com/search/docs/monitor-debug/security/social-engineering

So.....not all that helpful, but perhaps confirms some of the suspicions mentioned in previous comments? Zero guidance as to specifics, however. To be honest, even getting in contact with a person was difficult (I ended up calling the google domain abuse phone line listed on whois, which led to them creating a support ticket via email....)

**Comment by GodBleak** [comment]
I can't recommend this hack, as I'm certain I've seen some articles stating that it can actually be a cause of the issue, but I can't seem to find these articles again to cite them. So for those at their wits' end and willing to take that risk, it may be worth trying the following before giving up on Jellyfin.

I've moved Jellyfin behind an NGINX reverse proxy, and have been using it in conjunction with a script to automatically update the NGINX config to deny access to Jellyfin from any of the IPs Google has [listed as being used by Googlebot](https://developers.google.com/static/search/apis/ipranges/googlebot.json). I made this change on the first of January in a hope that Safe Browsing relies on Googlebot to make the decision of whether or not to flag the site and this would essentially prevent it from reaching any conclusion. So far, it seems to be working.

This will obviously hurt search rankings for your Jellyfin instance, but I think, for most, that's not going to be an issue. Also, I imagine there isn't overlap between the IPs Googlebot and other Google services (like Chromecast) use, but if there is, this would also affect Jellyfin's ability to use those services. 

If you're going to try this, and your domain is currently flagged, follow the appeal process and wait for the flag to be removed first, as Google will crawl your site as part of that process.

**Comment by chenichadowitz** [comment]
Interesting, I'd be curious to hear any feedback after running like that for a while. I'll mention that I'm already running Jellyfin behind an nginx reverse proxy, and that after the first or second instance I did try changing the tags mentioned previously. Since I have a (semi)active case open with support, I may leave it as is for a while and see if it gets flagged again without any further changes. If it does.....I may try the same thing if you've found that to be effective.

**Comment by iMartyn** [comment]
Folks, check your apex domains.  I am extremely annoyed at google suddenly deciding to do this as I have run loads of stuff from my domain, and each subdomain has a valid letsencrypt certificate, but apparently my apex domain was returning a redirect behind a self-signed certificate on my yunohost server.  People on mastodon pointed out that this is likely the cause for me.  My setup hasn't changed for that for years and we will see if this "fixes" the problem as I have followed the "report bad detection" flow.

**Comment by lednerg** [comment]
The second and final time I filed a report to Google about this, I mentioned that if it continued, I would have to get lawyers involved since their actions were negatively affecting my business (which they were). That was back in November and it's been fine ever since. I don't know for certain if that's what helped me, but it does make me think that Jellyfin should be seeking legal advice themselves if they are at all able, and this goes doubly for Emby since they can more easily claim this is affecting them financially. It could even be the kind of thing an attorney would assist you guys with at no charge, pro bono. What Google is doing is patently absurd, they know that they're messing with these products and they just don't care. Anyway, I'm sorry if this isn't the place for this.

**Comment by fange26** [comment]
> **Describe The Bug** A domain hosting Jellyfin is flagged by Google as a "Deceptive Site".
> 
> **Steps To Reproduce** Unknown
> 
> **System (please complete the following information):**
> 
> * Browser: Firefox, Chrome
> * Jellyfin Version: 10.8.5 (linuxserver/jellyfin:10.8.5-1-ls180)
> 
> **Additional Context** Google claims that https://example.tld/web/index.html
> 
> > attempts to trick users into doing something dangerous, such as installing unwanted software or revealing personal information.
> 
> I've appealed to Google twice now, but the domain continues to be flagged. This issue has been further documented on a few reddit posts:
> 
> * [Deceptive Site Ahead](https://www.reddit.com/r/jellyfin/comments/xqk01i/deceptive_site_ahead)
> * ["Deceptive site ahead" warning on my self hosted jellyfin web app](https://www.reddit.com/r/selfhosted/comments/xq162z/deceptive_site_ahead_warning_on_my_self_hosted/)

Google is flagging it as a deceptive site because the login being on the homepage throws a phishing flag in their system. You have to report that it is not deceptive to get the flag removed. 

you have to go to (https://search.google.com/search-console/security-issues) while signed into your google account and search the url, fix the issues with the site and then click the button for google to review the url. 

![google](https://user-images.githubusercontent.com/33208153/236641770-6d428a95-a34b-4606-92e9-cbd274d51a19.png)

**Comment by fange26** [comment]
@GodBleak 
WORKAROUND until Google removes flag for Jellyfin Server.

Android App not working. 
1. Go to GOOGLE PLAY STORE
2. Click on your PICTURE or AVATAR
3. Click PLAY PROTECT
4. Click the GEAR (Wheel or Cog) 
5. Turn off SCAN APPS WITH PLAY PROTECT
6. Clear the Cache for Jellyfin App and you the app should be able to accept your URL.

Google Chrome
1. Go to Settings
2. Go to Privacy and Security
3. Click Security
4. Select No Protection (not recommended)
5. Clear History 
6. go to jellfyin url.

**Comment by nielsvanvelzen** [comment]
FYI: The next version of the mobile Android app will disable safe browsing, so this issue won't happen with the app anymore.

**Comment by GodBleak** [comment]
> @GodBleak Google is flagging it as a deceptive site because the login being on the homepage throws a phishing flag in their system. You have to report that it is not deceptive to get the flag removed.
> 
> you have to go to (https://search.google.com/search-console/security-issues) while signed into your google account and search the url, fix the issues with the site and then click the button for google to review the url.
> 
> ![google](https://user-images.githubusercontent.com/33208153/236641770-6d428a95-a34b-4606-92e9-cbd274d51a19.png)

Have you read the thread @fange26? We're aware of this, however, it's not a solution.

**Comment by fange26** [comment]
It worked for me. In an hour my domain was unflagged.

**Comment by spottenn** [comment]
@fange26 If you read the full thread, lots of people have been getting their sites flagged, then unflagged by reporting the false positive, then flagged again days or weeks later. Would you mind coming back in a couple weeks and letting us know if yours got flagged again?

**Comment by fange26** [comment]
You can use the tool to remove your website from any Google scraping. I
made a request for them to remove the flag and to not scrape my website
anymore because it's a personal site, it's private.
You'll know that they've done it correctly if your jellyfish server no
longer shows up in Google searches. You can contact Google domain support
and they can walk you through accessing the page, claiming the website and
then requesting for the website not to be scraped.

**Comment by spottenn** [comment]
> I've moved Jellyfin behind an NGINX reverse proxy, and have been using it in conjunction with a script to automatically update the NGINX config to deny access to Jellyfin from any of the IPs Google has [listed as being used by Googlebot](https://developers.google.com/static/search/apis/ipranges/googlebot.json). I made this change on the first of January in a hope that Safe Browsing relies on Googlebot to make the decision of whether or not to flag the site and this would essentially prevent it from reaching any conclusion. So far, it seems to be working.
> 
> This will obviously hurt search rankings for your Jellyfin instance, but I think, for most, that's not going to be an issue. Also, I imagine there isn't overlap between the IPs Googlebot and other Google services (like Chromecast) use, but if there is, this would also affect Jellyfin's ability to use those services.

This seems the most promising workaround and since I've already got mine behind an NGINX reverse proxy, wouldn't be to hard to set up. We do use chromecast extensively though. Have you found out if it blocks Chromecast?

Would you mind sharing the script or how you set it up?

**Comment by spottenn** [comment]
>The MAIN TAKEAWAY from this is that you cannot get flagged if you are not in the search results. Only content that shows up in search results gets flagged.
My website isn't indexed, and is not and was not in the search results. I have had a robots.txt that requested my site not be indexed. However, my website is flagged. It got flagged today after my wife logged into my Jellyfin account on her iPhone in Chrome. I see that google does not recommend using a robots.txt to remove results, so maybe that affected it, but I think there are more variables at play.

**Comment by fange26** [comment]
@spottenn 
This is from Google. I followed these instruction to get my site permanently removed. 

Make your removal permanent
The Removals tool provides only a temporary removal of about six months. To remove content or a URL from Google search permanently:

1.Take one of the following actions to remove the page permanently:
   -Remove or update the content on your site (images, pages, directories) and make sure that your web server returns either a 404 (Not Found) or 410 (Gone) HTTP status code. Non-HTML files (like PDFs) should be completely removed from your server. (Learn more about [HTTP status codes](https://support.google.com/webmasters/answer/40132))
  -Block access to the content, for example by requiring a password.
  -Indicate that the page should not be indexed using the [noindex meta tag](https://developers.google.com/search/docs/crawling-indexing/block-indexing). This is less secure than the other methods.

Do not use [robots.txt](https://support.google.com/webmasters/answer/12818275) as a blocking mechanism.
If you blocked the page before removing your content permanently (step 1), unblock and then reblock the page. This clears the page from the index, if it was recrawled after blocking.

https://support.google.com/webmasters/answer/9689846#make_permanent&zippy=


<meta> tag To prevent all search engines that support the noindex rule from indexing a page on your site, place the following <meta> tag into the <head> section of your page:

**<meta name="robots" content="noindex">**

To prevent only Google web crawlers from indexing a page:

**<meta name="googlebot" content="noindex">**

https://developers.google.com/search/docs/crawling-indexing/block-indexing#meta-tag

**Comment by GiganteX** [comment]
My site has been flagged for quite some time. It's not in Google's search results. I have submitted multiple requests to google to remove my site from their warnings. I am getting incredibly frustrated. I hate Google so much.
I am running docker. I am not going to do any changes to my site that needs to be repeated every time I upgrade to a new container. Please, Google, just fix your shit.

**Comment by spottenn** [comment]
@alessandro-vitali @GiganteX 

Have either of you tried adding a login disclaimer? I had a similar issue with my Nextcloud domain until I changed the page title and did a couple other things to make it unique. I added a login disclaimer to Jellyfin yesterday and today my site is unblocked. We will see if it lasts.

**Comment by chrispalmeri** [comment]
This sounds like they are using image based machine learning these days, and can get data from users devices, not just bots. https://security.googleblog.com/2023/05/io-2023-android-security-and-privacy.html.html

My guess would be the logo is probably the main factor. Probably need something like https://github.com/jellyfin/jellyfin-web/issues/2335

**Comment by ulf5** [comment]
I added a login disclaimer and custom css, haven't had an issue since. Been months.

**Comment by alessandro-vitali** [comment]
@chrispalmeri thanks for the article, that's the first real clue to what could actually be going on here. Do you think it might be worth it to contact the developers behind this feature / the team responsible for it?

@spottenn @ulf5 how did you add the login disclaimer - using custom HTML/CSS on the login page or just with the built-in feature to display a message below the login fields? What did you add to the disclaimer - the info that this is a self-hosted instance with a link to the official jellyfin website/repo? I'm wondering how this might help with the ML-algorithm if they detect images - would it scan for links and detect the href to the 'correct' website?

**Comment by spottenn** [comment]
I also used the built-in feature. Without revealing the exact words, I said that: this is <my_domain>'s server. Not affiliated with jellyfin. Only use your credentials if trying to log in to <my_domain>.

**Comment by alessandro-vitali** [comment]
Thanks a lot for the info, I'll add the same to my login pages! As I'm also running Jellyseerr, Fallenbagel's fork of Overseerr on my domain, I might still run into issues though. Let's hope this does something.

@ulf5 Did you add custom CSS for this purpose - if so, what did you add?

**Comment by chrispalmeri** [comment]
I made a unique logo and was able to use it by adding custom css similar to this.

```css
.pageTitleWithDefaultLogo {
    background-image: url("data:image/png;base64,blahblahblah");
}
```

Replacing the url data with the output from something like https://www.atatus.com/tools/png-to-base64

It was eventually unflagged after several days, but I'm not sure yet if it will last. For some reason the issue came back for me pretty quickly earlier when I tried just adding a disclaimer.

**Comment by alessandro-vitali** [comment]
...and I'm back to square one again. Apparently adding a login disclaimer and blocking known Googlebot-IPs from accessing my domain is not yet enough. Fair enough, if I _was_ a bad actor it coudn't be that easy to bypass Google's great algorithm, _right_?

What about you @chrispalmeri? I'll try your solution next, I guess! The CSS-only solution is nice to have, I had already feared I'd have to change html in the web-folder - so that's a win :)

**Comment by thornbill** [comment]
The solution by @fange26 sounds the most realistic.

**Comment by alessandro-vitali** [comment]
Funny enough, I only actually got the "deceptive site" message / error for a few minutes, then it didn't appear anymore. Didn't get a notification through google search console either. Weird - it couldn't have been caching related or similar. Seems like it's "solved" for me, for now, at least.

I had implemented all of the steps above before, except password-protecting my site. Has been on my bucket list for some time though, as soon as I wrap my head around authelia to get some sort of SSO running. Thanks a lot for your support!

**Comment by PeeBeerBeach** [comment]
I also have this issue since recently.
Although my site is not indexed by Google, so i dont really understand why and how.

Other subfolder sites that run on the same domain work just  fine. It is just Jellyfin that comes up with the Deceptive Site warning.

**Comment by PeeBeerBeach** [comment]
Some of my SWAG conf files were outdated. I updated all of them and now the Deceptive site warning is gone for Jellyfin.
It still appears if I try to access an URL subfolder that does not exist on my webserver.
So at least in my case this does not seem to be a Jellyfin caused issue.

**Comment by sazz** [comment]
I unfortunately facing the issue as well. The really weird thing is that my server runs on my own local network. I have a constant internet connection with an opnsense as firewall and pi hole as dns blocker. So Google is not able to crawl the site. If I check on https://transparencyreport.google.com/safe-browsing/search I don't find my website either.

I'm running jellyfin on a virtual machine since three months now without any issues. I'm using a jellyfin-web frontend with some minor modifications and haven't updated my git clone since 25.03..2023.

Firefox is not having an issue. I tested some local services (each of them running in their own virtual machine) with Google but couldn't reproduce the issue.

I really have no clue what's going on.

**Comment by FelixLeChat** [comment]
The error disappeared for me a couple of days after making sure I has a A+ grade on [Security Headers](https://securityheaders.com/).

I use [caddy](https://caddyserver.com/), and mostly followed [this structure](https://gist.github.com/MatthewVance/876d1c0fe044c83e5bf80bd1a2dcd001).

**Comment by PeeBeerBeach** [comment]
Ok i was a bit too early. The deceptive site warning is now back on Jellyfin.
I can confirm that I have 6 red warnings at Security Headers scan. FelixLeChat might be right.

**Comment by PeeBeerBeach** [comment]
Ok i updated my ssl.conf and the security scan is now A+ and green.

Still the phising warning persists :(

**Comment by morkyy** [comment]
> The error disappeared for me a couple of days after making sure I has a A+ grade on [Security Headers](https://securityheaders.com/).
> 
> I use [caddy](https://caddyserver.com/), and mostly followed [this structure](https://gist.github.com/MatthewVance/876d1c0fe044c83e5bf80bd1a2dcd001).

If you can share exactly how you've set up your headers would be great! I've tried setting them up but unfortunately the content security policy ends up breaking Jellyfin for me on some browsers and also X-Frame-options break the LG TV jellyfin app for some weird reason....

**Comment by FelixLeChat** [comment]
> > The error disappeared for me a couple of days after making sure I has a A+ grade on [Security Headers](https://securityheaders.com/).
> > I use [caddy](https://caddyserver.com/), and mostly followed [this structure](https://gist.github.com/MatthewVance/876d1c0fe044c83e5bf80bd1a2dcd001).
> 
> If you can share exactly how you've set up your headers would be great! I've tried setting them up but unfortunately the content security policy ends up breaking Jellyfin for me on some browsers and also X-Frame-options break the LG TV jellyfin app for some weird reason....

Sure! Here is a [pastebin for my caddyfile](https://pastebin.com/FHvD8Rv9). Replace what is in between `[]` with your own information.

As for @PeeBeerBeach , it took me about 1-2 weeks with this config to be unflagged, but as the thread can demonstrate, your mile may vary for a lot of the proposed solutions.

**Comment by PeeBeerBeach** [comment]
Ok, thanks for the info. I guess i will sit and wait.
I will report back....

**Comment by PeeBeerBeach** [comment]
Still waiting.... ;)

**Comment by PeeBeerBeach** [comment]
Ok i can report back that the deceptive site warning disappeared for me.
I did what FelixLeChat suggested and also sent a message to Google asking to remove it.

Now the site  has no data at all in the Google search console, so it appears they just deleted all entries.

Let's see if it repears at some point....

**Comment by lhmngwy** [comment]
Yep, I can also confirm that getting an SSL rating of A, and reporting it to google via the link on the warning page fixed the issue within a few days.

I use Traefik and docker-compose and found this useful to improve my SSL rating: https://github.com/Khroners/Traefik-with-A-plus-on-SSL-Labs-Headers

**Comment by Tardnicus** [comment]
I also experienced this issue. Within hours of publishing my site to my domain, Google flagged it.

Fortunately, @FelixLeChat's instructions seemed to work for me. I updated my security headers as described, requested a review, and after some number of days, the warnings were removed.

For the record, I am using Caddy, and used the [helpful pastebin provided above](#issuecomment-1634279021).

I'll be sure to update this thread in case I get flagged again. I'm not _entirely_ convinced that the headers are the only component to this problem. I haven't yet updated my `og:` tags, so I wouldn't be surprised if I get flagged again for being "deceptive".

**Comment by coharness** [comment]
I am experiencing this issue as well. I am not using Caddy for my reverse proxy, but rather Synology's built-in service. Any advice on what steps to take for that setup?

**Comment by harrv** [comment]
> The MAIN TAKEAWAY from this is that you cannot get flagged if you are not in the search results. Only content that shows up in search results gets flagged.

@fange26 The steps you outlined in [your reply above](https://github.com/jellyfin/jellyfin-web/issues/4076#issuecomment-1537520193) are how to keep your site from appearing in Google search results. That part is correct. However, doing that will NOT help keep your site from being flagged as deceptive, as several other people have noted by their own experience. In my case I have kept my site out of Google's search result for years, and it was still flagged as deceptive.

Googles Search business and their API that browsers use to see if a site is on the deceptive list are two different things. If Google didn't flag any deceptive sites simply because the site maintainer used a robots metatag or marked their site as not searchable in the Search Console, then all nefarious websites would do that and Google's safe browsing service would be useless. The bots that index for search results honor the steps you outlined but the ones that look for deceptive sites certainly do not.

**Comment by harrv** [comment]
# All good for 5 months and counting!

Now that I've got your attention, at my house I run [Emby](https://emby.media) server, not Jellyfin. My Emby server was flagged as "deceptive" by Google a couple times back in March. I jumped through some hoops to make several changes to the branding, hoping that Google would no longer think my site was trying to impersonate another site just because it ran the same server software. Those changes seem to have worked, because my server has not been flagged again in the last five months.

Because Jellyfin branched from Emby when Emby [changed from open source to closed](/mediabrowser), I followed this issue closely and participated in the discussion here, knowing that the Jellyfin issue and the Emby one were very similar. Now that it's been five months without being re-flagged, I thought I'd write up a guide listing exactly what I changed and how.

**However**, as I was working on that guide, I noticed that several of the changes I made were no longer necessary because of some changes that Emby server has made sometime in the last few months--presumably because of all of the reports they had been getting about their users' sites being flagged by Google as deceptive.

So I fired up clean Emby and Jellyfin docker containers side-by-side to see more clearly what Emby has changed and whether Jellyfin has those changes too. (It doesn't yet.) I'm not entirely certain that Emby has made _all_ of the changes they need to make to prevent the issue, but the changes they have made so far are good ones and are some of the same changes that I made five months ago (although it was tricker for me to do it because I didn't have access to the Emby source).

## Recent changes made by Emby to address the issue

To show you some of the changes Emby has made, I will first show the relevant parts of the html source of Jellyfin's login page followed by Emby's login page. Five months ago Emby's was almost an exact match of Jellyfin's.


### Jellyfin's login page:
```html
<!doctype html>
<html class="preload">
<head>

<!-- ...irrelevant stuff... -->

  <link rel="manifest" href="64d966784cd77b03a79c.json">

<!-- ...irrelevant stuff... -->

  <meta name="application-name" content="Jellyfin">

<!-- ...irrelevant stuff... -->

  <meta property="og:title" content="Jellyfin">
  <meta property="og:site_name" content="Jellyfin">
  <meta property="og:url" content="http://jellyfin.org">
  <meta property="og:description" content="The Free Software Media System">

<!-- ...irrelevant stuff... -->
  
  <title>Jellyfin</title>

<!-- ...irrelevant stuff... -->
  
</head>

<body>

<!-- ...irrelevant stuff... -->
  
</body>
<html>
```

### Emby's login page:
```html
<!DOCTYPE html>
<html data-appversion="4.7.13.0" class="preload">
<head>

<!-- ...irrelevant stuff... -->

  <link rel="manifest" href="manifest.json">
  <meta name="description" content="c985b19dcd9d">

<!-- ...irrelevant stuff... -->

  <meta name="application-name" content="c985b19dcd9d">

<!-- ...irrelevant stuff... -->

  <meta property="og:title" content="c985b19dcd9d">
  <meta property="og:site_name" content="c985b19dcd9d">  
  <meta property="og:description" content="Energize your media.">

<!-- ...irrelevant stuff... -->

  <title>c985b19dcd9d</title>

<!-- ...irrelevant stuff... -->

</head>
<body>

<!-- ...irrelevant stuff... -->

</body>
</html>
```

### Important differences:

Jellyfin uses the name "Jellyfin" for the page `title`, the `application-name` meta tag, and several of the other meta tag properties. Several months ago when I was trying to resolve this issue, Emby was doing the same thing, except obviously using "Emby" in those spots rather than "Jellyfin". Today when I looked again, I saw that Emby is now using the server name in all of those spots. In this case I am using the linuxserver.io docker image and they use random hex digits for the server name. I found that I could change it by setting the server name in the Emby settings, so I made it "Steve's Server". Jellyfin also has a setting for server name, but they don't use it in the spots we're talking about.

When I made the changes to my server five months ago, I made all of these changes that Emby now does for you. Hopefully Jellyfin can make similar changes.

## Additional things I did that Emby has not yet changed

There are a few additional things that I changed on my server that may have helped me to reach the five month mark without being re-flagged that Emby has not yet addressed. I don't really know whether or not they are necessary to avoid being flagged, but these are the additional things I did:

1. I replaced all of the Emby logo image files with ones I made for the "Steve's Server" branding[^1]. That includes the `favicon.ico`, `splash.png` (Jellyfin calls it icon-transparent.png), and all of the `icon-<dimension>.png` files. I also changed `logowhite.png` and `logodark.png` files even though they don't appear on the login page and are therefore unlikely to be seen by the Google crawlers. I don't know what the Jellyfin equivalent of those files is.

2. Both Emby and Jellyfin have `manifest.json` files (although Jellyfin seems to randomize the name of many of their files, including that one). I left the reference to those files in the html posted earlier so you can see how they are referenced. If the Google web crawler decided to look inside the manifest files it would see a several references to Emby (or Jellyfin). I changed all of the Emby references inside `manifest.json` to also say "Steve's Server". The manifest file is also where you will find the `icon-<dimension>.png` files referenced that I mentioned above in item 1. One thing in the manifest file that I left alone are the pointers to the Android and iOS Emby apps.

3. There is a project called [Ombi](/Ombi-app/Ombi) that provides a request system that can be used with Plex, Emby and Jellyfin. I use that and it has a lot of the same branding issues, so I did some similar things with that. If I hadn't, I think Google very well may have re-flagged my server because the pages (including the login page) exposed in Ombi appear very much like the one on Ombi's demo site. Describing the (similar) changes I made to address this issue in Ombi is beyond the scope of what I'm describing here. I add this just to make the point that _any_ publicly accessable page on your server could cause you to get flagged, not just the ones exposed by Jellyfin.

[^1]: Obviously, Emby and Jellyfin are not going to make us all custom images, but it would be nice if they would give us a folder where we could put them and would use them if present.

### Some detail about how I went about it:

I'm no longer planning to write a full guide but I'll give a few details about the methods I used. There are a lot of ways of accomplishing it, but here's what I did...

I wanted to do the file replacements (such as the image files and the manifest file) in such a way that my changes wouldn't be lost whenever the server updated itself. Since I run Emby as a docker container I used Docker's volume mapping feature so that Docker would use certain files on the host machine rather than the ones in the Docker image.

However, the volume mapping trick is insufficient to make changes to the html title and meta tags because that content gets generated by source code and JavaScript, and I didn't want to go mucking with those, knowing that those kinds of changes would also get lost with Emby server updates. So I used [traefik](/traefik/traefik) as the ingress to my Emby container and added the [rewrite-body](/packruler/rewrite-body) plugin. I used that plugin to create a middleware to do regex replacements on the html headers I wanted to change. I know that some have used [nginx](/nginx/nginx) to accomplish something similar. See [this comment](https://emby.media/community/index.php?/topic/46707-how-to-change-name-of-title-of-bar-navigation/#comment-1239500) and the comment right before it in the Emby community support forums for slightly more detail.

**Comment by Manuel5cc** [comment]
I came to this issue because I am experimenting it. I see that is still ongoing since long ago, when it could be easily fixed by adding some options in the control panel. It's not a big issue since its a site that you obviously can trust if you built the server, but it is a little annoying.

So far I've tried adding a domain-level noindex tag and also asking google to remove it from the index with no luck.

**Comment by lhmngwy** [comment]
> I came to this issue because I am experimenting it. I see that is still ongoing since long ago, when it could be easily fixed by adding some options in the control panel. It's not a big issue since its a site that you obviously can trust if you built the server, but it is a little annoying.
> 
> So far I've tried adding a domain-level noindex tag and also asking google to remove it from the index with no luck.

Yes, it seems that even though I got unflagged temporarily when I got an A SSL rating, I soon became flagged again. It looks like we need a fix in jellyfin.

**Comment by metasmash** [comment]
Hello,

To give you some content, I am also running a whole media server stack in my machine (also with a nordvpn container from bubuntux)

So basically, the only active thread about deceptive websites are from jellyfin users. I've been running the whole stack with plex without nordvpn instead, and it worked well all long.

Multiple questions:

Are you guys using a VPN?
Or are we sure that Jellyfin has been considered as a phishing website or so?

Meanwhile, I'm also spending time to understand and solve this issues.It's been three weeks I've asked a review from google and didn't solve.

**Comment by ulf5** [comment]
I think what the flagging tells you is that it considers your instance a phishing threat to the Jellyfin instance hosted by Jellyfin. As I have explained previously in the thread, I made mine look different from the basic one by adding a disclaimer and some changes to the CSS. Then asked for a review, shortly after the flagging was removed and I haven't had an issue since.

Maybe the meta tag changes that Emby has made would stop Google from thinking that all instances are trying to phish, but I don't know if this is relevant to Emby as I don't know if they are hosting a demo instance.

**Comment by alanmccallum** [comment]
I haven't had this issue reoccur once my users and I stopped using Chrome.
My suspicion is that Google is using analytics and data from the users
browser to make the determination.

**Comment by metasmash** [comment]
> I think what the flagging tells you is that it considers your instance a phishing threat to the Jellyfin instance hosted by Jellyfin. As I have explained previously in the thread, I made mine look different from the basic one by adding a disclaimer and some changes to the CSS. Then asked for a review, shortly after the flagging was removed and I haven't had an issue since.
> 
> Maybe the meta tag changes that Emby has made would stop Google from thinking that all instances are trying to phish, but I don't know if this is relevant to Emby as I don't know if they are hosting a demo instance (but maybe it stops confusion between user hosted instances).

Indeed, I've read the thread. Everything make sense! I'll try this workaround with a script and I'll come back to you if it also worked for me 🫠

**Comment by metasmash** [comment]
Looks like my meta tags are successfully changed! It had been changed again when running javascript, but google crawlers shouldn't care about it since it doesn't execute javascript.

Here is what I've done:

### modify-meta.sh

```
#!/bin/bash

# Generate a random string of 32 characters
RANDOM_STRING=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)

# Replace meta tag content with the random string in index.html
sed -i 's/\(<meta name="[^"]*"\s*content="\)[^"]*"/\1'"$RANDOM_STRING"'"/g; s/\(<meta property="[^"]*"\s*content="\)[^"]*"/\1'"$RANDOM_STRING"'"/g' /jellyfin/jellyfin-web/index.html
```

`chmod +x modify_meta.sh`

### Dockerfile

```
# Use jellyfin/jellyfin:latest as base
FROM jellyfin/jellyfin:latest

# Copy our script to the container
COPY modify_meta.sh /modify_meta.sh

# Make the script executable
RUN chmod +x /modify_meta.sh

# Modify the entry point to execute our script before the main command
ENTRYPOINT ["/bin/bash", "-c", "/modify_meta.sh && exec /jellyfin/jellyfin"]
```
`docker build -t jellyfin-meta-replace .`

### docker-compose.yml


```
version: "3.9"

services:

  jellyfin:
    image: jellyfin-meta-replace
    container_name: jellyfin
    environment:
      - PUID=$PUID
      - PGID=$PGID
      - TZ=$TZ
    volumes:
      - ./library:/config
      - ${MNT}/transcodes:/config/transcodes
      - ${MNT}/tv:/data/tvshows
      - ${MNT}/movies:/data/movies
      - ${MNT}/cache:/cache
      - ./modify_jellyfin_meta.sh:/init/modify_jellyfin_meta.sh
    ports:
      - 8096:8096
    restart: unless-stopped
    labels:
     # Traefik conf
     
     #... other services
```


Result:

```

  <!-- HTML Meta Tags -->
  <title>86ssOc4eTLrkvrt8olokUgXNqp2MH2sR</title>
  <meta name="description" content="86ssOc4eTLrkvrt8olokUgXNqp2MH2sR">

  <!-- Facebook Meta Tags -->
  <meta property="og:url" content="SECRET">
  <meta property="og:type" content="website">
  <meta property="og:title" content="86ssOc4eTLrkvrt8olokUgXNqp2MH2sR">
  <meta property="og:description" content="86ssOc4eTLrkvrt8olokUgXNqp2MH2sR">
  <meta property="og:image" content="">

  <!-- Twitter Meta Tags -->
  <meta name="twitter:card" content="summary_large_image">
  <meta property="twitter:domain" content="SECRET">
  <meta property="twitter:url" content="SECRET">
  <meta name="twitter:title" content="86ssOc4eTLrkvrt8olokUgXNqp2MH2sR">
  <meta name="twitter:description" content="86ssOc4eTLrkvrt8olokUgXNqp2MH2sR">
  <meta name="twitter:image" content="">

  <!-- Meta Tags Generated via https://www.opengraph.xyz -->
        
```

If I get it well, this should correctly change all meta content of the website. I don't know if it is sufficient. Let's see how it goes!

**Comment by mikezila** [comment]
I'm also having this issue, though for me it doesn't seem to impact the usability of the Android app. We really need more controls to easily customize the landing page to prevent this.

**Comment by KaXaSA** [comment]
I had the same issue 2 weeks ago, I made some changes to the meta tags in the index page, also made some design changes, colors, sizes, etc. to make it more unique (not sure if it matters at all), and I also updated my caddyfile with the header stuff mentioned in this issue by the other users. Maybe it would be a good idea to update the [Jellyfin Caddy page](https://jellyfin.org/docs/general/networking/caddy/) ?

The weird thing is that on my PC, chrome (and other browsers) were not flagging the site at all (using the public URL), another user mentioned the issue, then I checked https://transparencyreport.google.com/safe-browsing and yep, google was actually flagging it as deceptive. 

Since at the same time I also changed the domain, I don't know if those changed alone would fix the problem, I guess I will have to wait and see, hopefully it won't get flagged again.

**Comment by mikezila** [comment]
Quick question. Of the people who have run into this, do you use the word "jellyfin" in your url anywhere? I found that I got hit with this right away when my instance was jellyfin.<my-unique-and-cool-name>.duckdns.org but when I changed it I have not been flagged again.

**Comment by metasmash** [comment]
> Quick question. Of the people who have run into this, do you use the word "jellyfin" in your url anywhere? I found that I got hit with this right away when my instance was jellyfin..duckdns.org but when I changed it I have not been flagged again.

Actually my website is named "stream.{{domain}}"

My whole domain name has been flagged as deceptive, which means that even my other deployed services are deceptive.

I've been running the exact same stack with plex a two year ago on a virtual private server, and no problems. I decided to switch to Jellyfin since the UI is simpler and open source code.

**Comment by KaXaSA** [comment]
> Quick question. Of the people who have run into this, do you use the word "jellyfin" in your url anywhere? I found that I got hit with this right away when my instance was jellyfin..duckdns.org but when I changed it I have not been flagged again.

I had no jellyfin in my URL, it was something like: https://name.domain-dns.net

**Comment by Echupk** [comment]
hey, after less than 2 months, i have been flagged again by Google Safe browser.
even if I chnaged the logo, disclaimer etc. My URL is TV.domain.fr
I saw that I put in the disclaimer that the website was not affiliated to jelyfin, i suppress the jellyfin mention, just in case.

I would like to know about the meta tags in the index file:
I'm running watchtowerr to update my containers, and jellyfin is updated automatically.
If iI change the tags in the index.html, will them still be changed when jellyfin is upgraded or not ? 

regards

**Comment by metasmash** [comment]
> hey, after less than 2 months, i have been flagged again by Google Safe browser. even if I chnaged the logo, disclaimer etc. My URL is TV.domain.fr I saw that I put in the disclaimer that the website was not affiliated to jelyfin, i suppress the jellyfin mention, just in case.
> 
> I would like to know about the meta tags in the index file: I'm running watchtowerr to update my containers, and jellyfin is updated automatically. If iI change the tags in the index.html, will them still be changed when jellyfin is upgraded or not ?
> 
> regards

I think it depends on the way you're updating them. If you have manually updated your jellyfin index.html, then it restores your jellyfin index.html based on the latest version. If you're using a script on your docker compose file to replace some fields according to regex rules, then you'll keep an index.html file updated according to the specified rules. 

Also, it's been 3 weeks I have published my new docker image with all meta tags changed, and my websites are still deceptive ahead.

**Comment by Echupk** [comment]
oh shit.
and I have a friend using it on a synology, (me on OMV), redirect with NPM like me.
He has never been flagged, but me yes...
and he never changed is index.html.
jist the logo and the banner at the beginnig and that's all

**Comment by metasmash** [comment]
Mh... At some point, I am wondering if the dns provider has something to do with this... Maybe can I ask you guys what dns provider are you using? I am under OVH currently. Maybe they have an automated report system or whatever flagging your websites due to policy or security issues?

**Comment by Echupk** [comment]
i was at duckdns first, an was flagged a lot, but then I learned that bots server use duckdns a lot and sometime domain where flags by mistake.
I changed recentrly to gandi.net (the same as my friend) but sill got flagged :/

**Comment by tcamargo** [comment]
My local/private jellyfin instance was flagged. After a couple of days, the warning went away without any action from my side. Now, big red screen is back.

**Comment by zaffron** [comment]
having the same issue. Got flagged within hours

**Comment by tcamargo** [comment]
> My local/private jellyfin instance was flagged. After a couple of days, warning went away without any action from my side. Now, big red screen is back. I don't remember when it was flagged for the first time. More than 2 months ago for sure.

And it was unflagged a couple of hours later (<24h)...

**Comment by PeeBeerBeach** [comment]
It is still working for me 2 month later. Check my previous posts to see what i did to solve it.

**Comment by yaoweiprc** [comment]
Same to me, I'm from China
![image](https://github.com/jellyfin/jellyfin-web/assets/6896642/0d8b6b8b-ec01-4ae0-ae15-5b5f2afe2a39)

**Comment by Raul824** [comment]
I started getting this same error. 
Have been running below configuration from 2 years without any issues, but suddenly today got this error.

Jellyfin running on docker.
Free domain from duckdns
nginx setup with subdomain jellyfin.

I am able to proceed to the website using visit this site in details but the jellyfin app on android keeps on crashing.
Is there any workaround for at-least android app to work.

**Comment by guhcampos** [comment]
I started getting this.

I have the feeling it has something to do with public DNS records pointing to private IP addresses. If you think about it, it's a very effective practice for a scammer as a man-in-the-middle attack: just drop a rogue service in someone's private network and use a lookalike public DNS to fool people into it. Since you're not hosting the page publicly, you can't be taken down by a hosting provider, then just make that service ship data to a lambda somewhere and profit from it. Might also be useful for controlling botnets, as they can use DNS for service discovery, a little bit like ASUS does with `asusrouter.com`.

Still, it's extremely frustrating that we can't have a way to "trust" this domain, whatever it is. It feels like Google taking our choice from us once again, and it makes me want to move off from Chrome.

**Comment by Eiziv** [comment]
Does anyone have a good workaround for this? Its really starting to annoy me

**Comment by guhcampos** [comment]
Well, there is one... Using another browser instead of Chrome.

**Comment by PeeBeerBeach** [comment]
Just read my comments. It works after what I have done

**Comment by martinbaines** [comment]
For me removing "jellyfin" from the path that redirects though the reverse proxy did it.
So the url of the form
**mysubdomain.maindomain.com/jellyfin** 
Cause the deceptive site message
but
**mysubdomain.maindomain.com/nomentionofsquashyfish**
goes through just fine.

**Comment by KaXaSA** [comment]
> ...
> I added the 'Login Disclaimer' message (Dashboard > General > Branding)
> 
> > This webpage (URL_HERE) is not affiliated with anyone, etc.
> 
> (or something similar)
> 
> Then I made some changes to index page, replaced any mention of 'jellyfin', removed the background-image from .splashLogo (jellyfin logo that shows up before page loads), also made some design changes, colors, sizes, etc. to make it more unique (not sure if it matters at all).
> 
> I also updated my caddyfile with the header stuff mentioned in this issue by the other users. 
> (got an A+ grade on [Security Headers](https://securityheaders.com)) 
> Maybe it would be a good idea to update the [Jellyfin Caddy page](https://jellyfin.org/docs/general/networking/caddy/) ?
> ...

After doing all that I had no issues for the past ~2 months, now I checked the [site status (google)](https://transparencyreport.google.com/safe-browsing) and I see this:

> **Some pages on this site are unsafe**
>  The site [redacted] contains harmful content, including pages that:
> 
>    > Try to trick visitors into sharing personal info or downloading software
> 
> Unsafe content might only appear on some pages of a website. Check the URL of the specific directory or webpage you want to visit for more detailed safety info. 

🤡 

I guess I will try to remove the Jellyfin favicon and the 'Jellyfin' page title. If I'm not mistaken, you have to edit main.jellyfin.bundle.js to change the title? Because changing the <title>Jellyfin</title> in the index page made no difference.

Then, if that's not enough and google block it again with the 'deceptive' message, I guess I will just ignore it and tell people to not use chrome 🤷‍♂️

**Comment by Allexio** [comment]
Can confirm I changed the url from jellyfin.example.com to jelly.example.com and it didn't get blocked anymore.
Also FYI other chromium-based browsers (such as Brave) work fine.

**Comment by dana23746** [comment]
> Can confirm I changed the url from jellyfin.example.com to jelly.example.com and it didn't get blocked anymore. Also FYI other chromium-based browsers (such as Brave) work fine.

same here.

**Comment by DX37** [comment]
Got blocked today. Again...

**Comment by metasmash** [comment]
> Looks like my meta tags are successfully changed! It had been changed again when running javascript, but google crawlers shouldn't care about it since it doesn't execute javascript.
> 
> Here is what I've done:
> 
> ### modify-meta.sh
> ```
> #!/bin/bash
> 
> # Generate a random string of 32 characters
> RANDOM_STRING=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
> 
> # Replace meta tag content with the random string in index.html
> sed -i 's/\(<meta name="[^"]*"\s*content="\)[^"]*"/\1'"$RANDOM_STRING"'"/g; s/\(<meta property="[^"]*"\s*content="\)[^"]*"/\1'"$RANDOM_STRING"'"/g' /jellyfin/jellyfin-web/index.html
> ```
> 
> `chmod +x modify_meta.sh`
> 
> ### Dockerfile
> ```
> # Use jellyfin/jellyfin:latest as base
> FROM jellyfin/jellyfin:latest
> 
> # Copy our script to the container
> COPY modify_meta.sh /modify_meta.sh
> 
> # Make the script executable
> RUN chmod +x /modify_meta.sh
> 
> # Modify the entry point to execute our script before the main command
> ENTRYPOINT ["/bin/bash", "-c", "/modify_meta.sh && exec /jellyfin/jellyfin"]
> ```
> 
> `docker build -t jellyfin-meta-replace .`
> 
> ### docker-compose.yml
> ```
> version: "3.9"
> 
> services:
> 
>   jellyfin:
>     image: jellyfin-meta-replace
>     container_name: jellyfin
>     environment:
>       - PUID=$PUID
>       - PGID=$PGID
>       - TZ=$TZ
>     volumes:
>       - ./library:/config
>       - ${MNT}/transcodes:/config/transcodes
>       - ${MNT}/tv:/data/tvshows
>       - ${MNT}/movies:/data/movies
>       - ${MNT}/cache:/cache
>     ports:
>       - 8096:8096
>     restart: unless-stopped
>     labels:
>      # Traefik conf
>      
>      #... other services
> ```
> 
> # Result:
> ```
> 
>   
>   <title>86ssOc4eTLrkvrt8olokUgXNqp2MH2sR</title>
>   <meta name="description" content="86ssOc4eTLrkvrt8olokUgXNqp2MH2sR">
> 
>   
>   <meta property="og:url" content="SECRET">
>   <meta property="og:type" content="website">
>   <meta property="og:title" content="86ssOc4eTLrkvrt8olokUgXNqp2MH2sR">
>   <meta property="og:description" content="86ssOc4eTLrkvrt8olokUgXNqp2MH2sR">
>   <meta property="og:image" content="">
> 
>   
>   <meta name="twitter:card" content="summary_large_image">
>   <meta property="twitter:domain" content="SECRET">
>   <meta property="twitter:url" content="SECRET">
>   <meta name="twitter:title" content="86ssOc4eTLrkvrt8olokUgXNqp2MH2sR">
>   <meta name="twitter:description" content="86ssOc4eTLrkvrt8olokUgXNqp2MH2sR">
>   <meta name="twitter:image" content="">
> 
>   
>         
> ```
> 
> If I get it well, this should correctly change all meta content of the website. I don't know if it is sufficient. Let's see how it goes!

The warning is now gone! Idk if this answer is related yet

**Comment by Echupk** [comment]
Since my last flag in august, I change the server name in the admin dashboard.
i change the warning on the login page (by the admin dashboard too)
and then i change the logo too, with a personnalized CSS


_@import url(https://cdn.jsdelivr.net/gh/prayag17/JellyFlix@latest/default.css);
/* Login Page Logo*/

.imgLogoIcon {

content: url(https://i.imgur.com/WDmHg5O.png) !important; }

/* Main Drawer Mobile Logo*/

.adminDrawerLogo img {

content: url(https://i.imgur.com/WDmHg5O.png) !important; }

/* Home Page Logo*/

.pageTitleWithLogo {

background-image: url(https://i.imgur.com/WDmHg5O.png) !important; }_


Since then, i never been flagged again.

**Comment by SethMacKayChandler** [comment]
I had this same issue, but was able to resolve it thanks to advice in this thread. To work around this issue I changed the base URL in Jellyfin's networking settings from the default 'jellyfin' to 'stream'. I also changed my reverse proxy (ngnix subfolder) location name to match.

**Comment by gsemet** [comment]
I confirm that just putting a base url works, i set to `/player`, it makes the warning from chrome disappear.

**Comment by betam4x** [comment]
Please note that a thread on hackernews (https://news.ycombinator.com/item?id=45675015) TODAY (10/22/2025) hypothesizes Gmail as a culprit.

The hypothesis is that Gmail catches the URLs in emails, and if they appear to mock specific branding/IP, they get added to safe browsing blocklists.

I am just adding this comment in case it helps. I do use Jellyfin, but only locally, so I have no horses in this race.

**Comment by jjarthur** [comment]
> The hypothesis is that Gmail catches the URLs in emails, and if they appear to mock specific branding/IP, they get added to safe browsing blocklists.

I don't think it's anything to do with Gmail, it's much more likely that simply visiting the site on a (probably) Chrome browser will send off some data about the site. 

I have this issue and my URL has never seen Gmail.
